### PR TITLE
Add config files for gpu-enabled app

### PR DESCRIPTION
**Pull request by jaguillette**
# Overview

This PR sets up GPU environments for use in HUIT OOD. It includes two new course setups that use the GPU queue, CS 1090b and BST 236. Each of these includes a `form.yml.erb` style file in the `local` directory to control its behavior, and includes a `bc_queue` option to select the `gpu` partition in our setup.

# Changes

There are two new app configurations in the `local` directory that use the `gpu` queue, specified by the `bc_queue` option hard-coded to the form. Note that the `bc_queue` option must be set under the `attributes` and `form` yml keys in order for it to work when creating a new GPU app.

This PR also changes the way that the app setup handles course-controlled conda environments. Rather than relying on a downstream spack environment into which `miniconda3` must be installed, this setup uses the global spack environment for `conda` to create a conda environment in the course shared folder using the `--prefix` option when creating the conda environment. This cuts down on setup time for new environments using this app configuration. There is an accompanying PR in our scripts repository to make this setup easier: https://github.com/Harvard-ATG/atg-ood-scripts/pull/7

Another change in this PR is the addition of permissions to apps for admin users. This is done in the Ruby section of the `local/*.yml.erb` files created by the PR. It adds a check for if the user is included in an admin group, bypassing the check for course group membership if so and granting access to the app by setting the `cluster` to `*`. The fallback is the existing check for course group membership. Since this change also introduces the possibility that an admin user will not have access to a conda environment in a shared course folder that they do not have direct access to, it adds a folder permissions check in the form of trying to `ls` the folder containing the conda environment. If this fails, it will be visible in the `output.log` file that the user does not have permission to access the relevant environment, rather than a more ambiguous failure state.

# Notes

When installing Python packages that are intended to take advantage of the GPU, like `pytorch` and `tensorflow`, it's necessary to run their install process on a GPU-enabled node. So when setting up a conda environment for the GPU queue, run the conda install process on a GPU node, ideally using one of our install scripts via sbatch using `-p gpu`.

# References

- [`conda create` reference](https://docs.conda.io/projects/conda/en/latest/commands/create.html)
- [Manage conda environments](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html)

### CODE REVIEW

File: local/bst236.yml.erb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+if arrays_have_common_element(userGroups, adminGroups)
+  cluster="*"
+else
+  # If the user is not in an admin group, check if they're in an authorized Canvas group
+  userCanvasGroups = userGroups.flat_map{ |str| str.scan(/^canvas(\d+)-\d+/) }.flatten
+  enabledGroups = [
+    # Cannot have other enabled groups if a course installation is in use
+    # This is because the course installation uses the course shared folder,
+    # which is not accessible to users outside of that course.
+    "150579" # COMPSCI 1090B: Data Science 2: Advanced Topics in Data Science
```
Comment (Tanvez) [inline local/bst236.yml.erb:26]: Minor typo you might have missed: `BST 236: Computing I`

**Review by Tanvez — APPROVED** [review]
This looks good to me! Thanks for the detailed write-up @jaguillette
